src/main/java/com/sonymobile/tools/gerrit/gerritevents/GerritHandler.java

```diff
@@ -183,6 +276,7 @@ private void post(Work work) {
         try {
             logger.trace("putting work on queue.");
             workQueue.put(work);
+            queueWork(work);
```

Comment:
Why double storage of the events? You seem to have removed most of the `workQueue` usage already.

Reply:
I will remove this completely, but then the getWorkQueue method should also be removed which is an API change.

Reply:
`executor.getQueue()` ?

Reply:
executor.getQueue() returns BlockingQueue<Runnable> while  the interface defines BlockingQueue<Work>

Reply:
You could then keep the `workQueue` reference to the queue you give the executor. They would then be the same instance but with the correct type. Or just cast it since we control what goes in it I guess, both ways are slightly ugly in their own ways :)

Reply:
Maybe we can make Work implement also Runnable.
Then we could cast `executor.getQueue()` to `BlockingQueue<Work>`

Reply:
Not very easy without breaking more compatibility stuff, the way it's designed with the 'perform(Coordinator)' method.
But the simple fix to keep it would be something like
```java
List<Work> l = new ArrayList<>();
for (Runnable r : executor.getQueue()) {
    if (r instanceof EventWorker) {
        l.add(((EventWorker)l).work);
    }
}
```
I think that would iterate normally without blocking, but needs testing.

There might be some callers that are expecting to get the reference to the actual queue, so the javadoc needs to be updated as well. But at least we keep binary compatibility and hopefully the deprecation warning will make it so no user uses it any more.